### ISSUE TITLE
Get faster/earlier feedback than pushing to `main`

### ISSUE THREAD

**Issue by s0enke** [superwerker/superwerker#136]
Currently, one needs to rely on their local test env (if any) to gain feedback before merging to `main`. This leads to trial and error in `main` and as a consequence many (maybe confusing and superfluous) release tags.

It's desired that one get's faster/earlier feedback before merging to `main` .

One idea is to test pull-request with the same infrastructure used for the `main` CodePipeline (see tests/buildspec.yaml).

We have to keep in mind that testing with MAVM grants full access to the AWS account so only trusted/reviewed code should get tested. For a start, it might be considered to restrict the access to start a PR test (which includes getting a fresh account from the MAVM) to those with "merge to main" permissions.

**Comment by yawn** [comment]
Related to #158 / more generally about development flow

**Comment by KiranGosavi** [comment]
Becky and I were checking the webhook filter documentation and found that we can filter by commit message. Could this work for our problem?
e.g. : 
```Triggers:
        Webhook: true
        FilterGroups:
          - - Type: EVENT
              Pattern: PULL_REQUEST_CREATED,PULL_REQUEST_UPDATED
         - Type: COMMIT_MESSAGE
            - Pattern: \[CodeBuild\]
          
 https://docs.aws.amazon.com/codebuild/latest/userguide/github-webhook.html

**Comment by s0enke** [comment]
> Becky and I were checking the webhook filter documentation and found that we can filter by commit message. Could this work for our problem?

Could you specify which problem you mean? Do you mean the 

> We have to keep in mind that testing with MAVM grants full access to the AWS account so only trusted/reviewed code should get tested. For a start, it might be considered to restrict the access to start a PR test (which includes getting a fresh account from the MAVM) to those with "merge to main" permissions.

aka "only pushes from members of the superwerker org should trigger a CodeBuild run" problem?

**Comment by KiranGosavi** [comment]
@s0enke yes, that was a solution proposal to our "only pushes from members of the superwerker org should trigger a CodeBuild run" problem, as we were exploring actor_id, code_commit etc trigger parameters in the last section. We tried with code commit message filter "[CodeBuild]" and its working fine.

**Comment by sbstjn** [comment]
All runs of the CodeBuild projected failed so far.

```
An error occurred (ParameterNotFound) when calling the GetParameter operation: 
An error occurred (ParameterNotFound) when calling the GetParameter operation: 
An error occurred (ParameterNotFound) when calling the GetParameter operation: 
An error occurred (ParameterNotFound) when calling the GetParameter operation: 
An error occurred (ParameterNotFound) when calling the GetParameter operation: 
An error occurred (ParameterNotFound) when calling the GetParameter operation: 
An error occurred (ParameterNotFound) when calling the GetParameter operation: 
```

**Comment by mavogel** [comment]
Yes, we need to revise this

**Comment by s0enke** [comment]
This might be a race condition: templates are not yet deployed and CloudFormation tries to read them from S3. But I did not look into it.
 
A fix for that might be to deploy the templates in the CodeBuild job?

**Comment by sbstjn** [comment]
Trying to fix it with #233

**Comment by Reb-B** [comment]
Every commit in a pull request is tested with the same infrastructure as our main pipeline.
